### PR TITLE
Alternate Icons selection

### PR DESCRIPTION
**Pull request by Gio2018**
## Summary
* This PR allows users to select a custom icon among the available ones, that can be found in Settings -> App Icon

## References 
* Branch name

## Implementation Details
* Add `SelectIconView`, a `SwiftUI` view that lets users select an icon among the available ones
* Add `SelectIconViewModel`, view model of the aforementioned view, responsible to set the selected icon
* Add `PocketAppIcon`, a type that describes an icon

> [!NOTE]
> The used icon is temporary and while we wait for other icons, we can use this one to try the feature out

## Test Steps
* Build/run then go to Settings and tap "App Icon"
* Select an icon that is not currently selected and verify that it's set as the app icon (see animation below)

## PR Checklist:
- [ ] Added Unit / UI tests
- [x] Self Review (review, clean up, documentation, run tests)
- [x] Basic Self QA

## Screenshots
<p align=center>

</p>

### CODE REVIEW

File: PocketKit/Sources/PocketKit/Settings/SettingsView.swift [hunk 1]
```diff
@@ -50,6 +50,18 @@ struct SettingsForm: View {
...
+                        ZStack {
+                            NavigationLink(destination: SelectIconContainerView()) {
+                                EmptyView()
+                            }
+                            .opacity(0.0)
+                            .buttonStyle(PlainButtonStyle())
+                            SettingsRowButton(title: Localization.Settings.AppIcon.title, trailingImageAsset: .chevronRight) {
+                                model.trackAccountManagementTapped()
+                            }
+                            .accessibilityIdentifier("account-management-button")
```
Comment (bassrock) [inline PocketKit/Sources/PocketKit/Settings/SettingsView.swift]:
```suggestion
                            .accessibilityIdentifier("app-icon-button")
```

File: PocketKit/Sources/PocketKit/Settings/SettingsView.swift [hunk 2]
```diff
@@ -50,6 +50,18 @@ struct SettingsForm: View {
...
                             .accessibilityIdentifier("original-view-toggle")
+                        // Custom implementation to hide the > arrow and let us use our own.
+                        ZStack {
+                            NavigationLink(destination: SelectIconContainerView()) {
+                                EmptyView()
+                            }
+                            .opacity(0.0)
+                            .buttonStyle(PlainButtonStyle())
+                            SettingsRowButton(title: Localization.Settings.AppIcon.title, trailingImageAsset: .chevronRight) {
+                                model.trackAccountManagementTapped()
```
Comment (bassrock) [inline PocketKit/Sources/PocketKit/Settings/SettingsView.swift]: Can we update to an analaytic event for app icon changing?

File: PocketKit/Sources/PocketKit/Settings/AppIcon/SelectIconView.swift [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                    List(PocketAppIcon.allCases) { appIcon in
+                        HStack(spacing: 16) {
+                            Image(appIcon.previewName)
+                                .resizable()
+                                .aspectRatio(contentMode: .fit)
+                                .frame(width: 60, height: 60)
+                                .cornerRadius(12)
+                            Text(appIcon.description)
+                            Spacer()
+                            CheckboxView(isSelected: viewModel.selectedAppIcon == appIcon)
```
Comment (bassrock) [inline PocketKit/Sources/PocketKit/Settings/AppIcon/SelectIconView.swift:29]: Let's add analytics for which app icon they select

We can do an engagement event and for the detail, put the icon name.

**Review by bassrock — APPROVED** [review]
Approved, but lets follow up with analytics please

File: PocketKit/Sources/PocketKit/Settings/SettingsView.swift (hunk 1 above)
Comment (Gio2018) [reply PocketKit/Sources/PocketKit/Settings/SettingsView.swift]: done

Comment (Gio2018) [reply PocketKit/Sources/PocketKit/Settings/SettingsView.swift]: done

File: PocketKit/Sources/PocketKit/Settings/SettingsView.swift (hunk 2 above)
Comment (Gio2018) [reply PocketKit/Sources/PocketKit/Settings/SettingsView.swift]: done

File: PocketKit/Sources/PocketKit/Settings/AppIcon/SelectIconView.swift (hunk 3 above)
Comment (Gio2018) [reply PocketKit/Sources/PocketKit/Settings/AppIcon/SelectIconView.swift:29]: done

**Pull request merged**